tests/phpunit/tests/test-switch-blog-rewrite-rules.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+if ( is_multisite() ) :
+
+	class Switch_Blog_Rewrite_Rules_Test extends PLL_Multisites_TestCase {
+		public function test_rewrite_rules_when_switching_blog() {
+			global $wp_rewrite;
+
+			$this->clean_up_filters();
+
+			$pll_admin = $this->get_pll_admin_env();
+			do_action_ref_array( 'pll_init', array( &$pll_admin ) );
+
+			/*
+			 * Test blog with Polylang activated and pretty permalink with language as directory.
+			 */
+			switch_to_blog( (int) self::$blog_with_pll_directory->blog_id );
+
+			$wp_rewrite->init();
+			$wp_rewrite->flush_rules();
+			$rules = $wp_rewrite->wp_rewrite_rules();
+
+			$this->assertNotEmpty( $rules );
+			$this->assertArrayNotHasKey( '(en)/?$', $rules );
+			$this->assertArrayHasKey( '(fr)/?$', $rules );
+			$this->assertArrayNotHasKey( '(fr)/(fr)/?$', $rules );
```

Comment:
I would not only check rules from the home group but also at least rules from another group like 'post' or 'category'. The same for the other sites lower.

Reply:
Done in [d0e3811](https://github.com/polylang/polylang/pull/1352/commits/d0e3811ae3bb8c19f3cb6564e0e118471a5dfbcb).